langstream-kafka-runtime/src/main/java/ai/langstream/kafka/runner/KafkaProducerWrapper.java

```diff
@@ -160,20 +165,30 @@ public CompletableFuture<?> write(Record r) {
                 if (forcedKeySerializer) {
                     key = r.key();
                 } else {
-                    if (keySerializer == null) {
-                        keySerializer = getSerializer(r.key().getClass(), keySerializers, true);
+                    Class<?> keyClass = r.key().getClass();
+                    if (keySerializer == null
+                            || !(Objects.equals(keyClass, cacheKeyForKeySerializer))) {
+                        keySerializer = getSerializer(keyClass, keySerializers, true);
+                        cacheKeyForKeySerializer = keyClass;
                     }
                     key = keySerializer.serialize(topicName, r.key());
+                    log.info("Chosen serializer for key  {} is {}", keyClass, keySerializer);
```

Comment:
I'd log this only when the serializer changes

Reply:
good catch.
it was a left over